k8s_dp/k8s_dp.p4

```diff
@@ -845,6 +847,69 @@ control k8s_dp_control(
                     }
                 }
 
+                if (istd.pass == (PassNumber_t) 0)
+                {
+                    if (IS_IPV4_TCP) {
+                            if (tcp_syn_flag_set(hdr.tcp.flags)) {
+                                    if (tx_balance_tcp.apply().hit) {
+                                            do_clb_pinned_flows_add_on_miss = true;
+                                            create_reverse_ct = true;
+                                            //save_to_meta_tcp(hdr, meta);
+                                            meta.src_ip = hdr.ipv4.src_addr;
+                                            meta.dst_ip = hdr.ipv4.dst_addr;
+                                            meta.src_port = hdr.tcp.src_port;
+                                            meta.dst_port = hdr.tcp.dst_port;
+                                            pinned_flows.apply();
+                                    }
+                            } else {
+                                    //save_to_meta_tcp(hdr, meta);
+                                    meta.src_ip = hdr.ipv4.src_addr;
+                                    meta.dst_ip = hdr.ipv4.dst_addr;
+                                    meta.src_port = hdr.tcp.src_port;
+                                    meta.dst_port = hdr.tcp.dst_port;
+                                    if (pinned_flows_reverse.apply().miss) {
```

Comment:
Does this call result in applying of miss action if there is a miss or does it only check whether there will be a miss on this table? If it's the former, then won't it end up adding entry to pinned_flow_reverse table, which is not what we want to do here?

I believe we used this logic here earlier as there was no direction support in DPDK. Now that we have direction support, can we simply use a check for direction to simplify this code? Same for UDP case.

Reply:
This part of code is same as before --- no changes in logic done here.
However, we do need to revisit the LB specific code --- mostly to untangle the processing for the two directions so that they can be placed at right place in the overall processing flow. But, that would have to be a separate commit. We should open a JIRA so that we don't lose track of it.

Reply:
Yes, I may have missed this part before or may be am confused again. If this code was working as expected before, then I guess this line is just a check and is not actually applying the miss action.
Yes, for untangling/simplification, it would be good to keep track so we don't miss it.

Reply:
Just wondering, this is CT "add_on_miss" code and not direction detection code.

Reply:
Yes, this is add-on-miss ... nothing to do with direction detection.